packages/headless/src/features/attached-results/attached-results-analytics-actions.test.ts

```diff
@@ -71,61 +91,117 @@ const resultParams = {
     source: 'example sourceName',
     collection: 'example collectionName',
     permanentid: 'example contentIDValue',
+    author: 'example author',
   }),
 };
 
 const testResult = buildMockResult(resultParams);
 
-describe('logCaseAttach', () => {
-  it('should log #logCaseAttach with the right payload', async () => {
-    const engine = buildMockInsightEngine({
-      state: buildMockInsightState({
-        search: buildMockSearchState({
-          results: [testResult],
-        }),
-        insightCaseContext: {
-          caseContext: {
-            Case_Subject: exampleSubject,
-            Case_Description: exampleDescription,
+describe('attached results analytics actions', () => {
```

Comment:
We are in the process of upgrading one core dependency in Headless (redux toolkit) and this way of doing tests unfortunately won't work anymore with the new version.

I'm currently in the process of migrating all tests, and you could already do the change in this PR while adding new tests.

Here is an example of the type of change that's needed: https://github.com/coveo/ui-kit/pull/3644/files#diff-152307c304f6b731b4f0bb4e7591cec659599180ab305f7d24bee53d0b5e5f81R1

* Change import from `'../../test/mock-engine'` to `'../../test/mock-engine-v2'`
* Test calling the action directly: `logCaseAttach(testResult)()(engine.dispatch, ()=> engine.state, {} as ThunkExtraArguments)` as opposed to going through the engine dispatch. You can mock anything in `ThunkExtraArguments` instead of an empty object if needed, but for most analytics actions and this type of test, this should not be necessary🤞.

This comment would apply for all the new test files added in this PR ! Thanks !